File: 10_laba/test/by/bntu/fitr/java/inGoslingsFootsteps/Lab10/model/logic/FlightControllerTest.java
package by.bntu.fitr.java.inGoslingsFootsteps.Lab10.model.logic;

import by.bntu.fitr.java.inGoslingsFootsteps.Lab10.model.entity.AirTicketOfficeContext;
import by.bntu.fitr.java.inGoslingsFootsteps.Lab10.model.entity.Customer;
import by.bntu.fitr.java.inGoslingsFootsteps.Lab10.model.exception.BookingException;
import by.bntu.fitr.java.inGoslingsFootsteps.Lab10.util.exception.InputException;
import org.junit.Test;

import java.io.FileNotFoundException;

import static org.junit.Assert.assertEquals;

public class FlightControllerTest {

    @Test
    public void getAvailableFlightOfNewCustomerByLength() throws FileNotFoundException, BookingException {
        Customer customer = new Customer("Ivan","ivan_ivanov");
        FlightController flightController = new FlightController();
        AirTicketOfficeContext context = AirTicketOfficeContext.getContext();
        AirTicketOfficeContext availableFlight = flightController.getAvailableFlight(customer);
        assertEquals(availableFlight.getFlights().length, context.sizeOfFlight());
    }

    @Test
    public void getAvailableFlightOfExistingCustomerByLength() throws FileNotFoundException, BookingException, InputException {
        CustomerController customerController = new CustomerController();
        int isActual = 1;
        Customer customer = customerController.addNewCustomerOrGetExisting("Matvey","matvey_anisovich");
        FlightController flightController = new FlightController();
        AirTicketOfficeContext availableFlight = flightController.getAvailableFlight(customer);
        assertEquals(availableFlight.getFlights().length, isActual);
    }

    @Test(expected = BookingException.class)
    public void notAvailableFlight() throws FileNotFoundException, BookingException, InputException {
        CustomerController customerController = new CustomerController();
        Customer customer = customerController.addNewCustomerOrGetExisting("Helena","lenka_bokshic");
        FlightController flightController = new FlightController();
        flightController.getAvailableFlight(customer);
    }

    @Test(expected = BookingException.class)
    public void notTicketBookedByCustomer() throws FileNotFoundException, BookingException, InputException {
        Customer customer = new Customer("Ivan","ivan_ivanov");
        FlightController flightController = new FlightController();
        flightController.getFlightsBookedByCustomer(customer);
    }

    @Test
    public void getTicketsBookedExistingCustomerByLength() throws FileNotFoundException, BookingException, InputException {
        CustomerController customerController = new CustomerController();
        int isActual = 2;
        Customer customer = customerController.addNewCustomerOrGetExisting("Helena","lenka_bokshic");
        FlightController flightController = new FlightController();
        AirTicketOfficeContext availableFlight = flightController.getFlightsBookedByCustomer(customer);
        assertEquals(availableFlight.getFlights().length, isActual);
    }
}

File: 10_laba/C##/AirTicketOffice/AirTicketOfficeTest/by/bntu/fitr/java/ingoslingsfootsteps/model/logic/FlightControllerTest.cs
using AirTicketOffice.by.bntu.fitr.java.ingoslingsfootsteps.model.entity;
using AirTicketOffice.by.bntu.fitr.java.ingoslingsfootsteps.model.logic;
using AirTicketOffice.by.bntu.fitr.java.ingoslingsfootsteps.model;
using NUnit.Framework;
using System;
using System.Collections.Generic;
using System.Text;
using AirTicketOffice.by.bntu.fitr.java.ingoslingsfootsteps.model.exception;

namespace AirTicketOfficeTest.by.bntu.fitr.java.ingoslingsfootsteps.model.logic
{
    class FlightControllerTest
    {
        [Test]
        public void getAvailableFlightOfNewCustomerByLength()
        {
            CustomerController customerController = new CustomerController();
            Customer customer = customerController.AddNewCustomerOrGetExisting("Ivan", "ivan_ivanov");
            FlightController flightController = new FlightController();
            AirTicketOfficeContext context = AirTicketOfficeContext.Context;
            AirTicketOfficeContext availableFlight = flightController.GetAvailableFlight(customer);
            Assert.AreEqual(availableFlight.Flights.Length, context.SizeOfFlight());
        }

        [Test]
        public void getAvailableFlightOfExistingCustomerByLength()
        {
            CustomerController customerController = new CustomerController();
            int isActual = 1;
            Customer customer = customerController.AddNewCustomerOrGetExisting("Matvey", "matvey_anisovich");
            FlightController flightController = new FlightController();
            AirTicketOfficeContext availableFlight = flightController.GetAvailableFlight(customer);
            Assert.AreEqual(availableFlight.Flights.Length, isActual);
        }

        [Test]
        public void notAvailableFlight()
        {
            CustomerController customerController = new CustomerController();
            Customer customer = customerController.AddNewCustomerOrGetExisting("Helena", "lenka_bokshic");
            FlightController flightController = new FlightController();
            Assert.Throws<BookingException>(() => flightController.GetAvailableFlight(customer));
        }

        [Test]
        public void notTicketBookedByCustomer()
        {
            CustomerController customerController = new CustomerController();
            Customer customer = customerController.AddNewCustomerOrGetExisting("Ivan", "ivan_ivanov");
            FlightController flightController = new FlightController();
            Assert.Throws<BookingException>(() => flightController.GetFlightsBookedByCustomer(customer));
        }

        [Test]
        public void getTicketsBookedExistingCustomerByLength()
        {
            CustomerController customerController = new CustomerController();
            int isActual = 2;
            Customer customer = customerController.AddNewCustomerOrGetExisting("Helena", "lenka_bokshic");
            FlightController flightController = new FlightController();
            AirTicketOfficeContext availableFlight = flightController.GetFlightsBookedByCustomer(customer);
            Assert.AreEqual(availableFlight.Flights.Length, isActual);
        }
    }
}
